types/prosemirror-model/index.d.ts

```diff
@@ -543,7 +553,12 @@ declare class ProsemirrorNode<S extends Schema = any> {
      * descend into a node when the callback returns `false`.
      */
     descendants(
-        f: (node: ProsemirrorNode<S>, pos: number, parent: ProsemirrorNode<S>) => boolean | null | undefined | void,
+        f: (
+            node: ProsemirrorNode<S>,
+            pos: number,
+            parent: ProsemirrorNode<S>,
+            index: number,
```

Comment:
The [official documentation](https://prosemirror.net/docs/ref/#model.Node.descendants) doesn't contain the fourth parameter (`index`), hence it is not part of the public API and should not be included in the type definitions. If you believe the official documentation to be wrong, please send a PR to [ProseMirror/prosemirror-model](https://github.com/).